crates/bindings/src/lib.rs

```diff
@@ -45,13 +42,610 @@ pub use timestamp::Timestamp;
+///
+/// These reducers cannot be called manually
+/// and may not have any parameters except for `ReducerContext`.
+///
+/// ### `#[spacetimedb::reducer(init)]`
+///
+/// This reducer is run the first time a module is published
+/// and any time the database is cleared.
+///
+/// If an error occurs when initializing, the module will not be published.
+///
+/// ### `#[spacetimedb::reducer(client_connected)]`
+///
+/// This reducer is run when a client connects to the SpacetimeDB module.
+/// Their identity can be found in the sender value of the `ReducerContext`.
+///
+/// If an error occurs in the reducer, the client will be disconnected.
+///
+/// ### `#[spacetimedb::reducer(client_disconnected)]`
+///
+/// This reducer is run when a client disconnects from the SpacetimeDB module.
+/// Their identity can be found in the sender value of the `ReducerContext`.
+///
+/// If an error occurs in the disconnect reducer,
+/// the client is still recorded as disconnected.
+///
+/// ### `#[spacetimedb::reducer(update)]`
+///
+/// This reducer is run when the module is updated,
+/// i.e., when publishing a module for a database that has already been initialized.
+///
+/// If an error occurs when updating, the module will not be published,
+/// and the previous version of the module attached to the database will continue executing.
+///
+/// # Scheduled reducers
+///
+/// In addition to life cycle annotations, reducers can be made **scheduled**.
+/// This allows calling the reducers at a particular time, or in a loop.
+/// This can be used for game loops.
+///
+/// The scheduling information for a reducer is stored in a table.
+/// This table has two mandatory fields:
+/// - A primary key that identifies scheduled reducer calls.
+/// - A [`ScheduleAt`] field that says when to call the reducer.
+/// Managing timers with a scheduled table is as simple as inserting or deleting rows from the table.
+///
+/// A [`ScheduleAt`] can be created from a [`spacetimedb::Timestamp`](crate::Timestamp), in which case the reducer will be scheduled once,
+/// or from a [`std::time::Duration`], in which case the reducer will be scheduled in a loop. In either case the conversion can be performed using [`Into::into`].
+///
+/// ```rust
+/// use spacetime::{table, reducer, Timestamp, ScheduleAt, Table}
+/// use std::time::Duration;
+/// use log::debug;
+///
+/// // First, we declare the table with scheduling information.
+///
+/// #[table(name = send_message_schedule, scheduled(send_message))]
+/// struct SendMessageSchedule {
+///     // Mandatory fields:
+///     // ============================
+///
+///     /// An identifier for the scheduled reducer call.
+///     #[primary_key]
+///     #[autoinc]
+///     scheduled_id: u64,
+///
+///     /// Information about when the reducer should be called.
+///     #[scheduled_at]
+///     scheduled_at: ScheduleAt,
+///
+///     // After the mandatory fields, any number of fields can be added.
```

Comment:
```suggestion
///     // In addition to the mandatory fields, any number of fields can be added.
```

Order doesn't matter. Field names don't matter either, actually, only type and attributes, but we don't need to mention that.